cscs-checks/libraries/magma/magma_checks.py

```diff
@@ -7,29 +7,36 @@
 import reframe.utility.sanity as sn
 
 
-@rfm.parameterized_test(['cblas_z'], ['zgemm'],
-                        ['zsymmetrize'], ['ztranspose'])
+@rfm.simple_test
 class MagmaCheck(rfm.RegressionTest):
-    def __init__(self, subtest):
-        self.valid_systems = ['daint:gpu', 'dom:gpu']
-        self.num_gpus_per_node = 1
-        self.sanity_patterns = sn.assert_found(r'Result = PASS', self.stdout)
+    subtest = parameter(['cblas_z', 'zgemm', 'zsymmetrize', 'ztranspose'])
+    valid_systems = ['daint:gpu', 'dom:gpu']
+    valid_prog_environs = ['builtin']
+    num_gpus_per_node = 1
+    prebuild_cmds = ['patch < patch.txt']
+    modules = ['magma']
+    maintainers = ['AJ', 'SK']
+    tags = {'scs', 'production', 'maintenance'}
 
-        self.prebuild_cmds = ['patch < patch.txt']
+    @run_after('init')
+    def set_build_system_opts(self):
         self.build_system = 'Make'
-        self.valid_prog_environs = ['builtin']
-        self.build_system.makefile = 'Makefile_%s' % subtest
+        self.build_system.makefile = 'Makefile_%s' % self.subtest
         # Compile with -O0 since with a higher level the compiler seems to
-        # optimise something away
+        # optimise stuff away
```

Comment:
@ajocksch, is it bad that the compiler is optimizing stuff away? Does it break the build?